SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorToolbar.cpp

```diff
@@ -996,21 +996,25 @@ void FSpatialGDKEditorToolbarModule::CloudDeploymentClicked()
 	USpatialGDKEditorSettings* SpatialGDKEditorSettings = GetMutableDefault<USpatialGDKEditorSettings>();
 	SpatialGDKEditorSettings->SpatialOSNetFlowType = ESpatialOSNetFlow::CloudDeployment;
 	SpatialGDKEditorSettings->bUseDevelopmentAuthenticationFlow = true;
-	USpatialGDKSettings* SpatialGDKSettings = GetMutableDefault<USpatialGDKSettings>();
-
-	FString DevAuthToken;
-	if (!SpatialCommandUtils::GenerateDevAuthToken(SpatialGDKSettings->IsRunningInChina(), DevAuthToken))
-	{
-		UE_LOG(LogSpatialGDKEditorToolbar, Error, TEXT("Failed to generate a development authentication token."));
-	}
-	SpatialGDKEditorSettings->DevelopmentAuthenticationToken = DevAuthToken;
-	SpatialGDKEditorSettings->SaveConfig();
-	SpatialGDKEditorSettings->SetRuntimeDevelopmentAuthenticationToken();
-
-	// Ensure we enable bUseDevelopmentAuthenticationFlow when using cloud deployment flow.
-	SpatialGDKSettings->bUseDevelopmentAuthenticationFlow = true;
-	SpatialGDKSettings->DevelopmentAuthenticationToken = DevAuthToken;
+	
 
+	AsyncTask(ENamedThreads::AnyBackgroundThreadNormalTask, [this]
+		{
+			FString DevAuthToken;
+			USpatialGDKSettings* GDKRuntimeSettings = GetMutableDefault<USpatialGDKSettings>();
+			if (!SpatialCommandUtils::GenerateDevAuthToken(GDKRuntimeSettings->IsRunningInChina(), DevAuthToken))
```

Comment:
We should only do the call to `GenerateDevAuthToken` in the async thread. Saving the result into the settings should be done on the game thread.